app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala

```diff
@@ -1878,6 +1882,30 @@ class SectionRenderingService(
     )
   }
 
+  private def getSmartStringWithInjectedSuffixes(
+    sString: SmartString,
+    formModelOptics: FormModelOptics[DataOrigin.Mongo]
+  )(implicit
+    sse: SmartStringEvaluator
+  ): SmartString = {
+    var lString: LocalisedString =
+      sString.localised(formModelOptics.formModelVisibilityOptics.booleanExprResolver.resolve(_))
+    var index: Long = 0
```

Comment:
Let's make the code more functional (immutable) by eliminating side effects. Can we use zipWithIndex and foldLeft instead of var?
`sString.allInterpolations.zipWithIndex.foldLeft(lString) {
  case (accumulatedString, (formCtx: FormCtx, index)) =>`